src/main/scala/cromwell/binding/Scope.scala

```diff
@@ -183,4 +174,37 @@ trait Scope {
    */
   def collectAllScatters = Scope.collectAllScatters(Seq(this), Nil)
 
-}
+  /*
+   * Calls and scatters are accessed frequently so this avoids traversing the whole children tree every time.
+   * Lazy because children are not provided at instantiation but rather later during tree building process.
+   * This prevents evaluation from being done before children have been set.
+   *
+   * FIXME: In a world where Scope wasn't monolithic, these would be moved around
```

Comment:
I said at standup today that you convinced me to wait until after the s/g stuff was done, and then i'm going in with the sledgehammer